src/types.rs

```diff
@@ -1,6 +1,30 @@
 use crate::Error;
 use derive_builder::Builder;
 
+use std::net::IpAddr;
+use ux::u24;
+
+pub(crate) const HEADER_LENGTH: usize = 16; // PRS_SOMEIP_00030
+pub(crate) const SD_HEADER_LENGTH: usize = 4;
+pub(crate) const SD_ENTRY_LENGTH: usize = 16;
+pub(crate) const IP4_SD_OPTION_LENGTH: usize = 12;
+pub(crate) const IP6_SD_OPTION_LENGTH: usize = 24;
+
+/// SomeIP message
+#[derive(Debug, PartialEq)]
+pub enum Message<'a> {
+    /// RPC Message
+    RpcMessage(Header, RpcPayload<'a>),
+    /// SD Message
+    SdMessage(Header, SdPayload),
```

Comment:
That's correct! Was probably driven by the use-case of parsing messages from a trace containing SD & RPC messages.. (btw. the "cookies" are only related to RPC via TCP.. and eg in C++ embedded stack just ignored ;-)

Reply:
". the "cookies" are only related to RPC via TCP)" 

makes the correct definition even worse:

```rust
enum Rpc {
  Rpc(&[u8]),  //  The name here is hard because there are events. This protocol is really weird.
  MagicCookieServer,
  MagicCookieClient,
}

enum Message {
  Rpc(Rpc),
  Sd(SdMessage), // I don't recall the exact name
}
```

this may result in [large_enum_variant](https://rust-lang.github.io/rust-clippy/master/#large_enum_variant)

Reply:
tbc!